### PR TITLE
Openapi

### PR DESCRIPTION
**Pull request by kcirtapfromspace**
# Note this is not working, thus draft.
Also was trying to get auto generated openapi docs. Though, admittedly i’m not clear on configuring the routes between the frontend & backend. 
This is rebased with the docker image caching  https://github.com/HHS/Head-Start-TTADP/pull/1970

## Description of change

Looking to add openapi documentation. 

- Serve api documentation on the site
- Autogenerate docs through tagging 

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-0

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -33,7 +35,7 @@ services:
     depends_on:
       - minio
   clamav-rest:
-    image: ajilaag/clamav-rest
+    image: kcirtapfromspace/clamav-rest
```
Comment (GarrettEHill) [inline docker-compose.yml:38]: I know this is yours, but I see that no one else is maintaining a cross platform version.
I think we need to reconfigure the clamav settings. As the import system is attempting to queue all the files its placing on s3 to be scanned. But they are ~75MB each.

File: package.json [hunk 2]
```diff
@@ -15,8 +15,8 @@
     "start:web": "node ./build/server/src/index.js",
     "start:worker": "node ./build/server/src/worker.js",
     "start:testingonly": "tsx watch ./src/testingOnly.js",
-    "server": "tsx watch src/index.ts",
-    "worker": "tsx watch src/worker.ts",
+    "server": "tsx watch --trace-warnings src/index.ts",
+    "worker": "tsx watch --trace-warnings src/worker.ts",
```
Comment (GarrettEHill) [inline package.json:19]: Will it get run with this flag in production or just in dev environments?

File: package.json [hunk 3]
```diff
@@ -52,28 +52,28 @@
...
+    "docker:deps": "docker compose run --rm backend yarn global add node-gyp && docker compose run --rm backend yarn install && docker compose run --rm frontend yarn global add node-gyp && docker compose run --rm frontend yarn install && docker compose run --rm testingonly yarn global add node-gyp && docker compose run --rm testingonly yarn install",
     "docker:reset": "./bin/reset-all",
-    "docker:start": "docker-compose up",
-    "docker:start:debug": "docker-compose --compatibility -f docker-compose.yml -f docker-compose.debug.yml up",
-    "docker:stop": "docker-compose down",
-    "docker:dbs:start": "docker-compose -f 'docker-compose.yml' up",
-    "docker:dbs:stop": "docker-compose -f 'docker-compose.yml' down",
+    "docker:start": "docker compose up",
+    "docker:start:debug": "docker compose --compatibility -f docker compose.yml -f docker compose.debug.yml up",
```
Comment (GarrettEHill) [inline package.json:58]: I think you went a little overzealous with ,your find/replace. the filename should still be docker-compose.yml

File: package.json [hunk 4]
```diff
@@ -282,6 +282,7 @@
     "sinon": "^15.0.0",
     "supertest": "^6.1.3",
     "tsx": "^3.12.2",
+    "turbo": "^1.9.6",
```
Comment (GarrettEHill) [inline package.json:285]: why turbo?

**Review by GarrettEHill — COMMENTED** [review]
I am trying to understand what this is all doing, seems like a mix of things

**Review by kcirtapfromspace — COMMENTED** [review]
> Will it get run with this flag in production or just in dev environments? 

This is just to build arm64 on m1 machines. open pr to clam for the to build armhttps://github.com/ajilach/clamav-rest/pull/37

File: docker-compose.yml (hunk 1 above)
Comment (kcirtapfromspace) [reply docker-compose.yml:38]: This is just to build arm64 on m1 machines. open pr to clam for the to build arm https://github.com/ajilach/clamav-rest/pull/37

**Review by kcirtapfromspace — COMMENTED** [review]
> Will it get run with this flag in production or just in dev environments? 

Probably better to have the conversation on https://github.com/HHS/Head-Start-TTADP/pull/1970 as that is focused on developer build systems

**Review by kcirtapfromspace — COMMENTED** [review]
> I think you went a little overzealous with ,your find/replace. the filename should still be docker-compose.yml  

Probably better to have the conversation on https://github.com/HHS/Head-Start-TTADP/pull/1970 as that is focused on developer build systems

**Comment by thewatermethod** [comment]
Confused about the aim here - we already serve the open API docs as part of the docker container network, is the idea to remove that container and serve them on cloud.gov?

**Comment by kcirtapfromspace** [comment]
> Confused about the aim here - we already serve the open API docs as part of the docker container network, is the idea to remove that container and serve them on cloud.gov?

Yeah, I'm thinking this is a would an easy method that serves load test tools the api docs url. If the docs are auto generated with jsDoc comments we shouldn't have to spell out yaml files. 

https://k6.io/blog/load-testing-your-api-with-swagger-openapi-and-k6/

**Pull request closed without merging**